### PR TITLE
feat: Include knative webhooks in readiness probes

### LINKED ISSUE #3867: Post-Install Webhook fails when installing Karpenter with 1 replica.
### Version

__Karpenter Version:__ v0.27.3

__Kubernetes Version:__ v1.26 (EKS)

### Expected Behavior

When installing the Karpenter Helm chart with `replicas: 1` it should work.

### Actual Behavior

Helm installs fails with the following error:

```
Error: failed post-install: warning: Hook post-install karpenter/templates/extra-manifests.yaml failed: Internal error occurred: failed calling webhook "defaulting.webhook.karpenter.k8s.aws": failed to call webhook: Post "https://karpenter.karpenter.svc:443/?timeout=10s": no endpoints available for service "karpenter"
```

### Steps to Reproduce the Problem

Install Karpenter via Helm using the following (redacted) values.yaml file:

```
replicas: 1
```

As soon as the replicas are configured to `2`, Karpenter can be installed as expected.

### Resource Specs and Logs

Karpenter Pod Logs:
[karpenter-655fcc864-tq56b.log](https://github.com/aws/karpenter/files/11425003/karpenter-655fcc864-tq56b.log)

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by ellistarn**
Fixes #3867

**Description**
Buddy PR to https://github.com/aws/karpenter-core/pull/334

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [ ] No

**Release Note**

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: charts/karpenter/values.yaml [hunk 1]
```diff
@@ -143,7 +143,7 @@ controller:
   sidecarVolumeMounts: []
   metrics:
     # -- The container port to use for metrics.
-    port: 8080
```
Comment (jonathan-innis) [inline charts/karpenter/values.yaml:146]: Did this change or was this just completely wrong?

File: go.mod [hunk 2]
```diff
@@ -7,7 +7,7 @@ require (
 	github.com/PuerkitoBio/goquery v1.8.1
 	github.com/avast/retry-go v3.0.0+incompatible
 	github.com/aws/aws-sdk-go v1.44.253
-	github.com/aws/karpenter-core v0.27.4-0.20230503190441-611e27dd0aef
```
Comment (jonathan-innis) [inline go.mod:10]: We should be bumping this to the latest HEAD right? Not the latest tagged version?

**Review by jonathan-innis — COMMENTED** [review]
Is there a way to ensure in testing that we are using these webhooks

File: charts/karpenter/values.yaml (hunk 1 above)
Comment (ellistarn) [reply charts/karpenter/values.yaml:146]: I was forced to change this due to the knative hardcoded port value. See the previous PR description.

Comment (jonathan-innis) [reply charts/karpenter/values.yaml:146]: Ah, I missed that. I assume that this value is propagated into our serviceMonitor. I think this is technically a breaking change for those that have manually created their service monitors or have other logic. Should we document this in our upgrade guide?

**Comment by ellistarn** [comment]
> Is there a way to ensure in testing that we are using these readiness/liveness probes on our webhooks?

I could write an e2e test that tries out a single replica install + immediately install a provisioner afterwards. Seems like a lot, though. If anything, WDYT catching this w/ a followon as part of the increased test coverage work?

File: charts/karpenter/values.yaml (hunk 1 above)
Comment (jonathan-innis) [reply charts/karpenter/values.yaml:146]: > Metrics are still scrapable at the karpenter service on port 8080

Reading the PR description, I don't get this line

**Comment by jonathan-innis** [comment]
> WDYT catching this w/ a followon as part of the increased test coverage work

Sounds good. As long as we are documenting it somewhere. Maybe we should create a mega-issue that captures test coverage in the Karpenter project?

File: charts/karpenter/values.yaml (hunk 1 above)
Comment (ellistarn) [reply charts/karpenter/values.yaml:146]: svc/karpenter forwards to the pod on a different port: 8080 -> 8000. Compatible with scrapers. 
```
k port-forward svc/karpenter -n karpenter 8080
Forwarding from 127.0.0.1:8080 -> 8000
Forwarding from [::1]:8080 -> 8000
Handling connection for 8080
```

Comment (ellistarn) [reply charts/karpenter/values.yaml:146]: In the service monitor, it's discovered by name:
```
  endpoints:
    - port: http-metrics
      path: /metrics
```

Comment (jonathan-innis) [reply charts/karpenter/values.yaml:146]: Got it, so it really only affects users who are referring to the port by name. Makes sense.

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Pull request merged**